streaming/base/storage/upload.py

```diff
@@ -620,17 +622,80 @@ def __init__(self,
                  progress_bar: bool = False) -> None:
         super().__init__(out, keep_local, progress_bar)
         self.client = self._create_workspace_client()
-        self.dbfs_path = self.remote.lstrip('dbfs:')  # pyright: ignore
-        self.check_folder_exists()
 
     def _create_workspace_client(self):
         try:
             from databricks.sdk import WorkspaceClient
+            return WorkspaceClient()
         except ImportError as e:
             e.msg = get_import_exception_message(e.name)  # pyright: ignore
             raise e
 
-        return WorkspaceClient()
+
+class DatabricksUnityCatalogUploader(DatabricksUploader):
+    """Upload file from local machine to Databricks Unity Catalog.
+
+    Args:
+        out (str | Tuple[str, str]): Output dataset directory to save shard files.
+
+            1. If ``out`` is a local directory, shard files are saved locally.
+            2. If ``out`` is a remote directory, a local temporary directory is created to
+               cache the shard files and then the shard files are uploaded to a remote
+               location. At the end, the temp directory is deleted once shards are uploaded.
+            3. If ``out`` is a tuple of ``(local_dir, remote_dir)``, shard files are saved in
+               the `local_dir` and also uploaded to a remote location.
+        keep_local (bool): If the dataset is uploaded, whether to keep the local dataset
+            shard file or remove it after uploading. Defaults to ``False``.
+        progress_bar (bool): Display TQDM progress bars for uploading output dataset files to
+            a remote location. Default to ``False``.
+    """
+
+    def __init__(self,
+                 out: Union[str, Tuple[str, str]],
+                 keep_local: bool = False,
+                 progress_bar: bool = False) -> None:
+        super().__init__(out, keep_local, progress_bar)
```

Comment:
I think we need a `self.check_folder_exists()` check here ?

Reply:
This is not currently supported for UC :(